Framework/API/src/AlgorithmNotification.cpp

```diff
@@ -0,0 +1,86 @@
+#include "MantidAPI/Algorithm.h"
```

Comment:
In general we ask the [All .cpp files must have a corresponding .h file](http://www.mantidproject.org/C%2B%2B_Coding_Standards#CPP_file_Headers). Putting this in a separate file works, but would it be clearer to move this into `Algorithm.cpp`?